### DISCUSSION TITLE
Add PR Number as a System Environment Variable

### DISCUSSION THREAD

**Discussion by tomagranate** [vercel/vercel#5785] [Ideas] [answered]
I'm trying to use a combination of Heroku's Review Apps and Vercel's Preview deployments, to get a full-stack preview deployment for each PR.

The issue I'm running into is that Heroku only allows the review apps to have a predictable URL that includes the PR number, while Vercel doesn't provide this an an environment variable. So I can't find a way to build my frontend with the right API url.

Any way you could add The Github PR Number as a System Environment Variable, so I could use it during my build step?

**Comment by tomagranate** [comment]
Just found this duplicate with a link to a package to help with the problem #5479

However, I definitely think native support is the far better way here.

**Comment by JClackett** [comment]
Pleaseee can we have this

**Comment by amyegan** [comment, marked as answer]
Pull request number is available now as a [system environment variable](https://vercel.com/docs/concepts/projects/environment-variables/system-environment-variables#system-environment-variables) called `VERCEL_GIT_PULL_REQUEST_ID`.

> The pull request id the deployment was triggered by. If a deployment is created on a branch before a pull request is made, this value will be an empty string.

**Reply by JClackett** [reply]
Omg 😍🥳🥳🥳
